src/Sulu/Component/Content/Document/Subscriber/RedirectTypeSubscriber.php

```diff
@@ -27,9 +28,23 @@ public static function getSubscribedEvents()
     {
         return [
             Events::METADATA_LOAD => 'handleMetadataLoad',
+            Events::PERSIST => 'handlePersist',
         ];
     }
 
+    public function handlePersist(PersistEvent $event)
+    {
+        $document = $event->getDocument();
+
+        if (!$document instanceof RedirectTypeBehavior) {
+            return;
+        }
+
+        if ($document->getRedirectTarget() === $document) {
+            throw new \InvalidArgumentException('You are not allowed to link a page to itself!');
+        }
+    }
```

Comment:
I hope that is ok. My first idea was to revert the use of the Metadata here, and use persist/hydrate again as it was before and currently is in other subscribers. But using the Metadata stuff reduces code a lot, and is much cleaner IMO. So I only used the persist event to check if the data is valid (which can't be done in the MetadataLoad event, because it just defines a mapping without the actual data being there).

/cc @wachterjohannes @chirimoya